mu/logic.py

```diff
@@ -411,9 +437,15 @@ def flash(self):
         logger.debug('Path to micro:bit: {}'.format(path_to_microbit))
         if path_to_microbit and os.path.exists(path_to_microbit):
             logger.debug('Flashing to device.')
-            hex_file = os.path.join(path_to_microbit, 'micropython.hex')
-            uflash.save_hex(micropython_hex, hex_file)
+            # Flash the microbit
+            uflash.flash(paths_to_microbits=[path_to_microbit],
+                         python_script=python_script,
+                         path_to_runtime=get_runtime_hex_path())
+            rt_hex_path = get_runtime_hex_path()
```

Comment:
```py
# Flash the microbit
uflash.flash(paths_to_microbits=[path_to_microbit],
                   python_script=python_script,
                   path_to_runtime=get_runtime_hex_path())
rt_hex_path = get_runtime_hex_path()
```
Do you see what i see? Two calls to `get_runtime_hex_path()`, 1 for the flashing, 2 for possible inclusion in the message.

What would probably make more sense is:
```py
# Flash the microbit
rt_hex_path = get_runtime_hex_path()
uflash.flash(paths_to_microbits=[path_to_microbit],
                   python_script=python_script,
                   path_to_runtime=rt_hex_path)
```